pkg/signup/service/signup_service.go

```diff
@@ -396,6 +392,13 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, provider ResourceProvider, u
 		return nil, nil
 	}
 
+	// Query BannedUsers to check the user has not been banned
+	banned, err := s.isUserEmailBanned(userSignup.Spec.IdentityClaims.Email)
```

Comment:
Alternatively you could just check the Complete condition reason here: https://github.com/codeready-toolchain/registration-service/blob/master/pkg/signup/service/signup_service.go#L438
Something like:
```
	} else if completeCondition.Reason == toolchainv1alpha1.UserSignupUserBannedReason {
		log.Info(nil, fmt.Sprintf("usersignup: %s is banned", userSignup.GetName()))
		// UserSignup is banned
		...
		return ...
	}
```
It would be fewer API requests and easier to test in unit tests I guess.

The question is what we actually want to return in the case of a banned user. If we return `nil` as you do in this PR then UI will treat it as a new user and will show user the "Signup" button which will fail if user clicks on it.
Not ideal.
Maybe we should return something like this instead:

```
		signupResponse.Status = signup.Status{
			Reason:               toolchainv1alpha1.UserSignupUserBannedReason,
		}
		return signupResponse, nil
```
That will be treated be UI as an existing users which is not yet provisioned. And I guess it will show a spinner...
Or we could use `toolchainv1alpha1.UserSignupPendingApprovalReason` and it will showed by UI as "Pending approval".

WDYT guys?

Reply:
> Alternatively you could just check the Complete condition reason

Good point, TBH checking the banned resource seemed easy enough, a bit less to test I agree. I can change it if it looks more obvious and there is low chance that the Condition will be dropped or changed.

> The question is what we actually want to return in the case of a banned user. If we return nil as you do in this PR then UI will treat it as a new user and will show user the "Signup" button which will fail if user clicks on it.

I think that we return `nil` response ( 404 Not found ) right now for banned users right ? So I didn't want to change the behaviour, just avoid logging the error when banned users try to login.

Reply:
BTW if we do return a specific response, like the ones @alexeykazakov suggested, would make the test more reliable since we don't have to check the `nil` response which may also be caused by usersignup not found.

Reply:
Good point about relying on UserSignup status. 👍 I like the idea of returning a response with `toolchainv1alpha1.UserSignupPendingApprovalReason` so that they just see the spinner and cannot keep retrying signup.

Reply:
> The current code does not return 404 if there is no MUR as I can see. It returns 500

You're right, but 500 is even less correct IMO. Since this is intentional and not a server error. I like the idea of returning a specific reason in case of user banned. So let's go with the Pending Approval if there is no objection.

Reply:
Please take a look at https://github.com/codeready-toolchain/registration-service/pull/404/commits/ceec4c6db058ca0245c1ce52e6e3e9610a641716 where I'm relying on the Usersignup condition and not anymore on the Banned resource, I've also updated the response to return `UserSignupPendingApprovalReason`.